lib/openstudio-standards/service_water_heating/create_water_heater.rb

```diff
@@ -0,0 +1,571 @@
+                                          coefficient_of_performance: 2.8,
+                                          electric_backup_capacity: 4500.0,
+                                          on_cycle_parasitic_fuel_consumption_rate: 0.0,
+                                          off_cycle_parasitic_fuel_consumption_rate: 0.0,
+                                          service_water_temperature: OpenStudio.convert(125.0, 'F', 'C').get,
+                                          service_water_temperature_schedule: nil,
+                                          set_peak_use_flowrate: false,
+                                          peak_flowrate: nil,
+                                          flowrate_schedule: nil,
+                                          water_heater_thermal_zone: nil,
+                                          service_water_loop: nil,
+                                          use_ems_control: false)
+      # create heat pump water heater
+      if heat_pump_type == 'WrappedCondenser'
+        hpwh = OpenStudio::Model::WaterHeaterHeatPumpWrappedCondenser.new(model)
+      elsif heat_pump_type == 'PumpedCondenser'
+        hpwh = OpenStudio::Model::WaterHeaterHeatPump.new(model)
+      end
+
+      # calculate tank height and radius
+      water_heater_capacity_kbtu_per_hr = OpenStudio.convert(water_heater_capacity, 'W', 'kBtu/hr').get
+      hpwh_vol_gal = OpenStudio.convert(water_heater_volume, 'm^3', 'gal').get
+      tank_height = (0.0188 * hpwh_vol_gal) + 0.0935 # linear relationship that gets GE height at 50 gal and AO Smith height at 80 gal
+      tank_radius = (0.9 * water_heater_volume / (Math::PI * tank_height))**0.5
+      tank_surface_area = 2.0 * Math::PI * tank_radius * (tank_radius + tank_height)
+      tank_ua = 3.9 # default ua assumption
+      u_tank = (5.678 * tank_ua) / OpenStudio.convert(tank_surface_area, 'm^2', 'ft^2').get
+      hpwh.setName("#{hpwh_vol_gal.round}gal Heat Pump Water Heater - #{water_heater_capacity_kbtu_per_hr.round(0)}kBtu/hr")
+
+      # set min/max HPWH operating temperature limit
+      hpwh_op_min_temp_c = OpenStudio.convert(45.0, 'F', 'C').get
+      hpwh_op_max_temp_c = OpenStudio.convert(120.0, 'F', 'C').get
+
+      if heat_pump_type == 'WrappedCondenser'
+        hpwh.setMinimumInletAirTemperatureforCompressorOperation(hpwh_op_min_temp_c)
+        hpwh.setMaximumInletAirTemperatureforCompressorOperation(hpwh_op_max_temp_c)
+        # set sensor heights
+        if hpwh_vol_gal <= 50.0
+          hpwh.setDeadBandTemperatureDifference(0.5)
+          h_ue = (1 - (3.5 / 12.0)) * tank_height # in the 4th node of the tank (counting from top)
+          h_le = (1 - (10.5 / 12.0)) * tank_height # in the 11th node of the tank (counting from top)
+          h_condtop = (1 - (5.5 / 12.0)) * tank_height # in the 6th node of the tank (counting from top)
+          h_condbot = (1 - (10.99 / 12.0)) * tank_height # in the 11th node of the tank
```

Comment:
should this be 10.5?

Reply:
I think it's correct. The comment may have been copied and pasted without being updated. It's from the residential hpxml code. @jmaguire1 I think was the original author of this and may know.